pkg/drift/drift.go

```diff
@@ -0,0 +1,137 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package drift
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/abcxyz/guardian/pkg/drift/assets"
+	"github.com/abcxyz/guardian/pkg/drift/iam"
+	"github.com/abcxyz/guardian/pkg/drift/terraform"
+)
+
+func Process(ctx context.Context, organizationID int64, bucketQuery string) error {
+	assetsClient, err := assets.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("error initializing assets Client: %w", err)
+	}
+	folders, err := assetsClient.GetHierarchyAssets(ctx, organizationID, "cloudresourcemanager.googleapis.com/Folder")
+	if err != nil {
+		return fmt.Errorf("unable to get folders: %w", err)
+	}
+	projects, err := assetsClient.GetHierarchyAssets(ctx, organizationID, "cloudresourcemanager.googleapis.com/Project")
+	if err != nil {
+		return fmt.Errorf("unable to get folders: %w", err)
+	}
+	buckets, err := assetsClient.GetBuckets(ctx, organizationID, bucketQuery)
+	if err != nil {
+		return fmt.Errorf("error fetching terraform state GCS buckets: %w", err)
+	}
+	fmt.Printf("Fetching IAM for %d Folders and %d Projects\n", len(folders), len(projects))
+
+	gcpIAM, err := getActualGCPIAM(ctx, organizationID, folders, projects)
+	if err != nil {
+		return fmt.Errorf("error determining GCP IAM: %w", err)
+	}
+	fmt.Printf("Fetching terraform state from %d Buckets\n", len(buckets))
+	tfIAM, err := getTerraformStateIAM(ctx, organizationID, folders, projects, buckets)
+	if err != nil {
+		return fmt.Errorf("error determining Terraform State: %w", err)
+	}
+	fmt.Printf("Found %d gcp IAM entries\n", len(gcpIAM))
+	fmt.Printf("Found %d terraform IAM entries\n", len(tfIAM))
+
+	clickOpsChanges := difference(gcpIAM, tfIAM)
+	missingTerraformChanges := difference(tfIAM, gcpIAM)
+
+	fmt.Printf("Found %d Click Ops Changes\n", len(clickOpsChanges))
+	fmt.Printf("Found %d Missing Terraform Changes\n", len(missingTerraformChanges))
+
+	return nil
+}
+
+func getActualGCPIAM(
+	ctx context.Context,
+	organizationID int64,
+	folders []assets.HierarchyNode,
+	projects []assets.HierarchyNode,
+) (map[string]iam.AssetIAM, error) {
+	client, err := iam.NewClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error initializing iam Client: %w", err)
+	}
+
+	m := make(map[string]iam.AssetIAM)
+	for _, f := range folders {
+		fIAM, err := client.GetIAMForFolder(ctx, f, "folders")
+		if err != nil {
+			return nil, fmt.Errorf("unable to get folder IAM for folder with ID '%d' and name '%s': %w", f.ID, f.Name, err)
+		}
+		for _, i := range fIAM {
+			m[iam.CreateURI(i, organizationID)] = i
+		}
+	}
+	for _, p := range projects {
+		pIAM, err := client.GetIAMForProject(ctx, p, "projects")
+		if err != nil {
+			return nil, fmt.Errorf("unable to get project IAM for project with ID '%d' and name '%s': %w", p.ID, p.Name, err)
+		}
+		for _, i := range pIAM {
+			m[iam.CreateURI(i, organizationID)] = i
+		}
+	}
+
+	return m, nil
+}
+
+func getTerraformStateIAM(
```

Comment:
Nit: Go functions are not named "Get" or "Set", so this should be `terraformStateIAM`.